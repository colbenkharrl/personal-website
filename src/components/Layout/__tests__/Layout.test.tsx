import { render, screen } from '@testing-library/react';
import * as stories from '../__stories__/Layout.stories';
import { composeStories } from '@storybook/react';

const { Basic } = composeStories(stories);

jest.mock('../../Header', () => ({
  Header: () => <p>Header</p>,
}));
jest.mock('../../Footer', () => ({
  Footer: () => <p>Footer</p>,
}));

it('Layout matches snapshot (regression test)', async () => {
  const content = 'Test blog content';

  render(
    <div data-testid="Layout">
      <Basic />
    </div>,
  );

  await screen.findByText(content);

  expect(screen.getByTestId('Layout')).toMatchSnapshot(
    'Full component snapshot.',
  );
});
