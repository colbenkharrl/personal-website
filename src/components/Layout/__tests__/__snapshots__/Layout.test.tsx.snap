// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches snapshot (regression test): Full component snapshot. 1`] = `
<div
  data-testid="Layout"
>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
  </script>
  <div
    class="fixed inset-0 flex justify-center sm:px-8"
  >
    <div
      class="flex w-full max-w-7xl lg:px-8"
    >
      <div
        class="w-full bg-white ring-1 ring-zinc-100 dark:bg-zinc-900 dark:ring-zinc-300/20"
      />
    </div>
  </div>
  <div
    class="relative flex w-full flex-col"
  >
    <p>
      Header
    </p>
    <main
      class="flex-auto"
    >
      Test blog content
    </main>
    <p>
      Footer
    </p>
  </div>
</div>
`;
